File: solutions/csharp/perfect-numbers/1/PerfectNumbers.cs
using System;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number <= 0) throw new ArgumentOutOfRangeException();
        if (number == 1) return Classification.Deficient;
        int fs = 1;
        for (int n = 2; n<number; n++) {
            if (number % n == 0) fs += n;
        }
        if (fs==number) return Classification.Perfect;
        if (fs > number) return Classification.Abundant;
        return Classification.Deficient;
    }
}

File: solutions/fortran/perfect-numbers/1/perfect_numbers.f90

module perfect_numbers
  implicit none

contains

  character(len=9) function classify(num)
    integer, intent(in) :: num
    integer asum, i
    if (num < 1) then
      classify = "ERROR"
      return
    end if  
    if (num == 1) then
      classify = "deficient"
      return
    end if  
    asum=1
    do i = 2,num-1
      if (mod(num,i) == 0) then
        asum=asum+i
      end if
    end do
    if (asum == num) then
      classify = "perfect"
    else if (asum > num) then
      classify = "abundant"
    else 
      classify = "deficient"
    end if
    return
  end function

end module
